operation/client_handler.go

```diff
@@ -165,6 +166,8 @@ func (client *clientHandler) handleBasicCommand(command architecture.Command) ar
 		}
 	case architecture.TOUCH:
 
+	case architecture.QUIT:
+		client.conn.Close()
```

Comment:
There's one more complication here since you need to return from `startSession`. Otherwise though you close the connection the `go routine` that runs the client does not exit. This would create a resource leak.